src/core/Akka.Remote.Tests/RemoteDeathWatchSpec.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Actor.Dsl;
+using Akka.Configuration;
+using Akka.Event;
+using Akka.TestKit;
+using Akka.TestKit.TestEvent;
+using Akka.Util;
+using Xunit;
+
+namespace Akka.Remote.Tests
+{
+    public class RemoteDeathWatchSpec : AkkaSpec
+    {
+        private static readonly Config _config = ConfigurationFactory.ParseString(@"
+            akka {
+                actor {
+                    provider = ""Akka.Remote.RemoteActorRefProvider, Akka.Remote""
+                    deployment {
+                        /watchers.remote = ""akka.tcp://other@localhost:2666""
+                    }
+                }
+                remote.retry-gate-closed-for = 1 s
+                remote.initial-system-message-delivery-timeout = 3 s
+                remote.netty.tcp {
+                    hostname = ""localhost""
+                        port = 0
+                }
+            }
+        ");
+
+        private readonly ActorSystem _other;
+
+        public RemoteDeathWatchSpec() : base(_config)
+        {
+            _other = ActorSystem.Create("other",
+                ConfigurationFactory.ParseString(@"akka.remote.netty.tcp.port=2666").WithFallback(_config));
+        }
+
+        protected override void BeforeTermination()
+        {
+            var mute = EventFilter.Warning(pattern: new Regex("received dead letter.*Disassociate")).Mute();
+            Sys.EventStream.Publish(mute);
+        }
+
+        protected override void AfterTermination()
+        {
+            _other.Terminate().Wait(TimeSpan.FromSeconds(20));
+        }
+
+        [Fact]
+        public void Must_receive_Terminated_when_system_of_deserialized_ActorRef_is_not_running()
+        {
+            var probe = CreateTestProbe();
+            Sys.EventStream.Subscribe(probe.Ref, typeof(QuarantinedEvent));
+            var rarp = RARP.For(Sys).Provider;
+            // pick an unused port (not going to use a socket address generator here; just a port not used by either actor system)
+            int port = rarp.DefaultAddress.Port.Value;
+            while (port == rarp.DefaultAddress.Port.Value || port == 2666)
+                port = ThreadLocalRandom.Current.Next(1, 65535);
+            // simulate de-serialized ActorRef
+            var @ref = rarp.ResolveActorRef($"akka.tcp://OtherSystem@localhost:{port}/user/foo/bar#1752527294");
+            Action<IActorDsl> act = dsl =>
```

Comment:
We do have DSL specs: https://github.com/akkadotnet/akka.net/blob/dev/src/core/Akka.Tests/Actor/ActorDslSpec.cs